### PR TITLE
refactor: address generator and parser

### PR DESCRIPTION
**Pull request by shaojunda**

### CODE REVIEW

File: lib/ckb/address_parser.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    def initialize(address)
+      @address = address
+    end
+
+    def parse
+      decoded_prefix, data = ConvertAddress.decode(address)
+      format_type = data[0].unpack("H*").first
+
+      case format_type
+      when "01"
```
Comment (ashchan) [inline lib/ckb/address_parser.rb]: Reuse ‘SHORT_FORMAT’?

Comment (shaojunda) [reply lib/ckb/address_parser.rb]: thanks, updated: https://github.com/nervosnetwork/ckb-sdk-ruby/pull/209/commits/837502db0a9605e5291128e901f24fd0a72aace9

**Comment by classicalliu** [comment]
Conflicts

**Comment by shaojunda** [comment]
> Conflicts

fixed

**Comment by shaojunda** [comment]
please review again @classicalliu

File: lib/ckb/address.rb [hunk 2]
```diff
@@ -2,145 +2,92 @@
...
-      prefix = prefix(mode: mode)
-      format_type = Utils.to_hex(format_type)[2..-1].rjust(2, '0')
-      raise InvalidFormatTypeError.new("Invalid format type") unless TYPES[1..-1].include?(format_type)
-      raise InvalidArgsTypeError.new("Args should be a hex string") unless CKB::Utils.valid_hex_string?(args)
+      return generate_full_payload_address unless CKB::ScriptHashType::TYPE == script.hash_type && CKB::Utils.hex_to_bin(script.args).bytesize == 20
 
-      payload = [format_type].pack("H*") + CKB::Utils.hex_to_bin(code_hash) + CKB::Utils.hex_to_bin(args)
-
-      CKB::ConvertAddress.encode(prefix, payload)
+      if(SystemCodeHash::SECP256K1_BLAKE160_SIGHASH_ALL_TYPE_HASH == script.code_hash)
```
Comment (classicalliu) [inline lib/ckb/address.rb]:
```suggestion
      if SystemCodeHash::SECP256K1_BLAKE160_SIGHASH_ALL_TYPE_HASH == script.code_hash
```

**Pull request merged**